Snowflake.Data.Tests/UnitTests/ArrowResultSetTest.cs

```diff
@@ -0,0 +1,144 @@
+﻿/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Apache.Arrow;
+using Apache.Arrow.Ipc;
+using NUnit.Framework;
+using Snowflake.Data.Core;
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    [TestFixture]
+    class ArrowResultSetTest
+    {
+        private static int s_rowCount = 10;
+        private RecordBatch _recordBatch;
+        private ArrowResultSet _arrowResultSet;
+
+        [Test]
+        [Ignore("ArrowResultSetTest")]
+        public void SFResultSetArrowTestDone()
+        {
+            // Do nothing - test progress marker
+        }
+        
+        [SetUp]
+        public void BeforeTest()
+        {
+            s_rowCount = 10;
+            _recordBatch = new RecordBatch.Builder()
+                .Append("Col_Int32", false, col => col.Int32(array => array.AppendRange(Enumerable.Range(1, s_rowCount))))
+                .Build();
+            var responseData = PrepareResponseData(_recordBatch);
+            var sfStatement = PrepareStatement();
+
+            _arrowResultSet = new ArrowResultSet(responseData, sfStatement, new CancellationToken());
+        }
+        
+        [Test]
+        public void TestNext()
+        {
+            var column = (Int32Array)_recordBatch.Column(0);
+            for (var i = 0; i < s_rowCount; ++i)
+            {
+                Assert.IsTrue(_arrowResultSet.Next());
+                Assert.AreEqual(column.GetValue(i), _arrowResultSet.GetValue<Int32>(0));
```

Comment:
Maybe we can test for instance comparing to `i`? Now you don't know if you really test the next row, if the value is still the same.

Reply:
The test tests Next() function, I'll remove comparison of values.